core/src/main/java/io/micronaut/core/io/service/ServiceScanner.java

```diff
@@ -187,6 +189,20 @@ private void findMicronautMetaServiceConfigs(BiConsumer<URI, String> consumer) t
             uniqueURIs.add(uri);
         }
 
+        if(uniqueURIs.isEmpty()) {
```

Comment:
Not 100% sure if this emtpy-check is correct. For JLink it is, which was the usecase. In case of combining classpath and modulepath this might need to be revisited, although I wonder if people would do that.